### PR TITLE
Docsite: update "Migrating Ansible content to a different collection" info

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
Docsite: update "Migrating Ansible content to a different collection" info

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
`docs/docsite/rst/dev_guide/developing_collections.rst`

### CODE REVIEW

**Comment by Andersson007** [comment]
@gundalow @felixfontein maybe anything else to add / change according to the current experience?

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 1]
```diff
@@ -543,8 +543,12 @@ To migrate content from one collection to another, if the collections are parts
 #. When the next major release comes:
 
   * remove the module/plugin from the source collection
-  * add ``redirect`` to the corresponding entry in ``meta/runtime.yml``
+  * remove its unit and integration tests, specific module utils, and documentation fragments if exist from the source collection. Be sure, they are not used by other modules/plugins and tests
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: > Be sure, they are not used by other modules/plugins and tests

How can we be sure, is that only for `tests/integration/targets/setup_*` tests?

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 2]
```diff
@@ -543,8 +543,12 @@ To migrate content from one collection to another, if the collections are parts
...
+  * add a changelog fragment containing entries for ``removed_features`` and ``breaking_changes``
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: Can we link to Felix's PRs as examples?

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 1 above)
Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: Yes

Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: `grep` usually helps

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 2 above)
Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: Thought of this, ok, will do, thanks!

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 1 above)
Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: i'll reformulate this a little bit

Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: i removed this sentence, it's, imo, unnecessary

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 3]
```diff
@@ -542,9 +542,13 @@ To migrate content from one collection to another, if the collections are parts
 #. Deprecate the module/plugin with ``removal_version`` scheduled for the next major version in ``meta/runtime.yml`` of the source collection. The deprecation must be released after the copied content has been included in a release of the target collection.
 #. When the next major release comes:
 
-  * remove the module/plugin from the source collection
-  * add ``redirect`` to the corresponding entry in ``meta/runtime.yml``
+  * remove the module/plugin from the source collection and its symlink stored in ``plugin/modules`` directory
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
  * remove the module/plugin from the source collection and its symlink stored in ``plugin/modules`` directory if appropriate (mainly when removing from ``community.general`` and ``community.network``)
```

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 4]
```diff
@@ -542,9 +542,13 @@ To migrate content from one collection to another, if the collections are parts
...
+  * remove its unit and integration tests, specific module utils, and documentation fragments if exist from the source collection
+  * add a changelog fragment containing entries for ``removed_features`` and ``breaking_changes``; refer to the fragment in this `pull request <https://github.com/ansible-collections/community.general/pull/1304>`_ as an example
+  * add ``redirect`` to the corresponding module/plugin's entry in ``meta/runtime.yml``
+  * add ``redirect`` for the removed module utils and documentation fragments
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
  * add ``redirect`` to the corresponding module/plugin's entry in ``meta/runtime.yml``
    * in particular, add ``redirect`` for the removed module utils and documentation fragments if applicable
```

Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: @felixfontein , committed, thanks!

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 5]
```diff
@@ -542,9 +542,13 @@ To migrate content from one collection to another, if the collections are parts
 #. Deprecate the module/plugin with ``removal_version`` scheduled for the next major version in ``meta/runtime.yml`` of the source collection. The deprecation must be released after the copied content has been included in a release of the target collection.
 #. When the next major release comes:
 
-  * remove the module/plugin from the source collection
-  * add ``redirect`` to the corresponding entry in ``meta/runtime.yml``
+  * remove the module/plugin from the source collection and its symlink stored in ``plugin/modules`` directory if appropriate (mainly when removing from ``community.general`` and ``community.network``)
+  * remove its unit and integration tests, specific module utils, and documentation fragments if exist from the source collection
```
Comment (dmsimard) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
  * remove its unit and integration tests, specific module utils, and documentation fragments if there are any in the source collection
```

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 6]
```diff
@@ -542,9 +542,13 @@ To migrate content from one collection to another, if the collections are parts
...
+  * add a changelog fragment containing entries for ``removed_features`` and ``breaking_changes``; refer to the fragment in this `pull request <https://github.com/ansible-collections/community.general/pull/1304>`_ as an example
+  * add ``redirect`` to the corresponding module/plugin's entry in ``meta/runtime.yml``
+    * in particular, add ``redirect`` for the removed module utils and documentation fragments if applicable
   * remove ``removal_version`` from there
+  * remove related entries from tests/sanity/ignore.txt files if exist
```
Comment (dmsimard) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
  * remove related entries from ``tests/sanity/ignore.txt`` files if exist
```

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 7]
```diff
@@ -542,9 +542,13 @@ To migrate content from one collection to another, if the collections are parts
...
```
Comment (dmsimard) [inline docs/docsite/rst/dev_guide/developing_collections.rst:559]:
```suggestion
* remove requirements that are no longer required in ``tests/unit/requirements.txt``
```

**Comment by dmsimard** [comment]
LGTM, just minor things and a note about unit test requirements

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 6 above)
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
  * remove related entries from tests/sanity/ignore.txt files if exist
  * remove changelog fragments for removed content that are not yet part of the changelog (i.e. do not modify `changelogs/changelog.yaml` and do not delete files mentioned in it)
```

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 7 above)
Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst:559]: @dmsimard thanks for the good suggestions, now it's better! Committed.

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 6 above)
Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: @felixfontein thanks, added

**Comment by Andersson007** [comment]
ready_for_review

**Comment by Andersson007** [comment]
ready_for_review

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 8]
```diff
@@ -542,9 +542,15 @@ To migrate content from one collection to another, if the collections are parts
 #. Deprecate the module/plugin with ``removal_version`` scheduled for the next major version in ``meta/runtime.yml`` of the source collection. The deprecation must be released after the copied content has been included in a release of the target collection.
 #. When the next major release comes:
```
Comment (dericcrago) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: Which `next major release` is this talking about?

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 9]
```diff
@@ -542,9 +542,15 @@ To migrate content from one collection to another, if the collections are parts
...
 
-  * remove the module/plugin from the source collection
-  * add ``redirect`` to the corresponding entry in ``meta/runtime.yml``
+  * remove the module/plugin from the source collection and its symlink stored in ``plugin/modules`` directory if appropriate (mainly when removing from ``community.general`` and ``community.network``)
+  * remove its unit and integration tests, specific module utils, and documentation fragments if there are any in the source collection
+  * add a changelog fragment containing entries for ``removed_features`` and ``breaking_changes``; refer to the fragment in this `pull request <https://github.com/ansible-collections/community.general/pull/1304>`_ as an example
+  * add ``redirect`` to the corresponding module/plugin's entry in ``meta/runtime.yml``
+  * in particular, add ``redirect`` for the removed module utils and documentation fragments if applicable
   * remove ``removal_version`` from there
```
Comment (dericcrago) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: are both of these points talking about the `meta/runtime.yml` file? Would it be helpful to either explicitly call that out or indent them?

maybe something like this even?

```
* changes to ``meta/runtime.yml``
  * add ``redirect`` to the corresponding module/plugin's entry
    * in particular, add ``redirect`` for the removed module utils and documentation fragments if applicable
  * remove ``removal_version``
```

**Review by dericcrago — COMMENTED** [review]
helpful updates @Andersson007, thank you!

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 8 above)
Comment (felixfontein) [reply docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
#. When the next major release of the old collection is prepared:
```

Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: @dericcrago thanks for the question! @felixfontein thanks for clarifying! committed

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 9 above)
Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: @dericcrago , good suggestion, done, thanks!

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 10]
```diff
@@ -540,11 +540,19 @@ To migrate content from one collection to another, if the collections are parts
...
+  * remove its unit and integration tests, specific module utils, and documentation fragments if there are any in the source collection
+  * add a changelog fragment containing entries for ``removed_features`` and ``breaking_changes``; refer to the fragment in this `pull request <https://github.com/ansible-collections/community.general/pull/1304>`_ as an example
+  * change ``meta/runtime.yml``:
+
+    * add ``redirect`` to the corresponding module/plugin's entry
+    * in particular, add ``redirect`` for the removed module utils and documentation fragments if applicable
+    * remove ``removal_version`` from there
+  * remove related entries from ``tests/sanity/ignore.txt`` files if exist
+  * remove changelog fragments for removed content that are not yet part of the changelog (in other words, do not modify `changelogs/changelog.yaml` and do not delete files mentioned in it)
+  * remove requirements that are no longer required in ``tests/unit/requirements.txt``
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
  * remove requirements that are no longer required in ``tests/unit/requirements.txt``, ``tests/requirements.yml`` and ``galaxy.yml``
```

Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: Done

**Comment by Andersson007** [comment]
cc @acozine @samccann

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 11]
```diff
@@ -540,11 +540,19 @@ To migrate content from one collection to another, if the collections are parts
...
 #. Copy content from the source (old) collection to the target collection.
 #. Deprecate the module/plugin with ``removal_version`` scheduled for the next major version in ``meta/runtime.yml`` of the source collection. The deprecation must be released after the copied content has been included in a release of the target collection.
-#. When the next major release comes:
-
-  * remove the module/plugin from the source collection
-  * add ``redirect`` to the corresponding entry in ``meta/runtime.yml``
-  * remove ``removal_version`` from there
+#. When the next major release of the old collection is prepared:
+
+  * remove the module/plugin from the source collection and its symlink stored in ``plugin/modules`` directory if appropriate (mainly when removing from ``community.general`` and ``community.network``)
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: Whatever term we use in the main intro (line 543 above), we should use throughout - so either "old collection" everywhere or "source collection" everywhere - I like "old collection" a bit better.
```suggestion
  * remove the module/plugin from the old collection
  * remove the symlink stored in ``plugin/modules`` directory if appropriate (mainly when removing from ``community.general`` and ``community.network``)
```

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 12]
```diff
@@ -540,11 +540,19 @@ To migrate content from one collection to another, if the collections are parts
...
+  * remove its unit and integration tests, specific module utils, and documentation fragments if there are any in the source collection
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: I think splitting this out so it reads like a checklist makes it easier to tell if I've done everything in the list:
```suggestion
  * remove related unit and integration tests
  * remove specific module utils
  * remove specific documentation fragments if there are any in the old collection
```

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 13]
```diff
@@ -540,11 +540,19 @@ To migrate content from one collection to another, if the collections are parts
...
+  * add a changelog fragment containing entries for ``removed_features`` and ``breaking_changes``; refer to the fragment in this `pull request <https://github.com/ansible-collections/community.general/pull/1304>`_ as an example
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: I had to read this twice to figure out what it meant . . . not sure how to express this better, maybe:
```suggestion
  * add a changelog fragment containing entries for ``removed_features`` and ``breaking_changes``; you can see an example of a changelog fragment in this `pull request <https://github.com/ansible-collections/community.general/pull/1304>`_ 
```

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 14]
```diff
@@ -540,11 +540,19 @@ To migrate content from one collection to another, if the collections are parts
...
+  * change ``meta/runtime.yml``:
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: More specifics would help here - change meta/runtime.yml in the old collection? in the new collection? in ansible/ansible?

**Review by acozine — COMMENTED** [review]
Thanks @Andersson007 this is a much more detailed instruction set now. I see a couple of ways it could be made even better.

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 11 above)
Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: I'm not sure `old` is better than `source` technically BUT, agreed, it's clearer

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 13 above)
Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: LGTM

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 14 above)
Comment (Andersson007) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: fixed, thanks!

**Comment by Andersson007** [comment]
ready_for_review

**Comment by acozine** [comment]
Thanks @Andersson007. This makes the page clearer and more complete than it was before. We can make more improvements over time.

**Comment by Andersson007** [comment]
@acozine @felixfontein @gundalow @dmsimard @dericcrago thank you for collaborating! :)

**Pull request merged**